src/hooks/walletDetect/index.ts

```diff
@@ -81,26 +81,52 @@ export const useIsFarcaster = () => {
   return isFarcaster;
 };
 
+/**
+ * A hook that detects if the application is running inside a Binance mini app.
+ *
+ * @returns A boolean indicating if the app is running inside a Binance mini app
+ */
+export const useIsBinance = () => {
+  const [isBinance, setIsBinance] = useState(false);
+
+  useEffect(() => {
+    const checkBinance = () => {
+      if (typeof window !== "undefined" && window.bn) {
+        setIsBinance(true);
```

Comment:
Was this `window.bn` mechanism tested? How can we know it will work? I can only see them in the mini-program product APIs which is a different product.